packages/actor-init-sparql/test/HttpServiceSparqlEndpoint-test.ts

```diff
@@ -870,3 +871,84 @@ describe('HttpServiceSparqlEndpoint', () => {
     });
   });
 });
+
+const variantsDefault = [{ type: 'application/json', quality: 1 },
+  { type: 'simple', quality: 1 },
+  { type: 'application/sparql-results+json', quality: 1 },
+  { type: 'application/sparql-results+xml', quality: 1 },
+  { type: 'table', quality: 1 },
+  { type: 'tree', quality: 0.9 },
+  { type: 'stats', quality: 1 },
+  { type: 'application/trig', quality: 1 },
+  { type: 'application/n-quads', quality: 0.7 },
+  { type: 'text/turtle', quality: 0.6 },
+  { type: 'application/n-triples', quality: 0.3 },
+  { type: 'text/n3', quality: 0.2 },
+  { type: 'application/ld+json', quality: 0.9 }];
+
+describe('determineMediaType', () => {
+  let instance: any;
+  let request: any;
+
+  beforeEach(async() => {
+    instance = new HttpServiceSparqlEndpoint({});
+    request = makeRequest();
+  });
+
+  function makeRequest() {
+    request = stringToStream('default_test_request_content');
+    request.url = 'url_sparql';
+    request.headers = { accept: 'application/json,application/trig,simple,stats' };
+    request.method = 'GET';
+    return request;
+  }
+
+  it('should return a null mediatype when given */* accept header'
+    , async() => {
+      request.url = 'url_undefined_query';
+      request.headers = { accept: '*/*' };
+      expect(instance.determineMediaType(request, variantsDefault)).toEqual(null);
+    });
+
+  it('should return a null mediatype when no accept header'
+    , async() => {
+      request.url = 'url_undefined_query';
+      request.headers = {};
+      expect(instance.determineMediaType(request, variantsDefault)).toEqual(null);
+    });
+
+  it('should choose default mediatype if it is present in accept', async() => {
+    request.headers = { accept: 'application/json,application/trig,simple,stats' };
+    instance.parseBody = jest.fn(() => Promise.resolve('test_result'));
+    request.method = 'POST';
+    expect(instance.determineMediaType(request, variantsDefault)).toEqual('application/json');
+  });
+
+  it('should choose second preferred mediatype if it is present in accept', async() => {
+    request.headers = { accept: 'application/trig,simple,stats' };
+    instance.parseBody = jest.fn(() => Promise.resolve('test_result'));
+    request.method = 'POST';
+    expect(instance.determineMediaType(request, variantsDefault)).toEqual('simple');
+  });
+
+  it('should choose the next preferred variant available in accept', async() => {
+    request.headers = { accept: 'application/trig,stats' };
+    instance.parseBody = jest.fn(() => Promise.resolve('test_result'));
+    request.method = 'POST';
+    expect(instance.determineMediaType(request, variantsDefault)).toEqual('stats');
+  });
+
+  it('should choose the preferred variant with different q values', async() => {
```

Comment:
I suspect you forgot to add the actual q-values as specified here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept

Reply:
Variants are passed with their quality to the handleRequest function which is why i did not need to add them . But I'll add tests which start from run then to make it parse the mediatypes by itself instead of passing it to the function.

Reply:
But these tests here don't go via `handleRequest`, right?
So those variants are not used here.

I would suggest having some tests with header in the form of `{ accept: 'application/n-quads;q=0.5,tree,application/n-triples;q=0.8,application/ld+json;q=0.2' }`

Reply:
handleRequest would call the determineMediaType with the variants passed as arguments. So it would be the same thing.

Reply:
Sure, but in this block we are specifically testing `determineMediaType`, so we should test exactly that. From the point of view of `determineMediaType`, `handleRequest` should not even exist, so all relevant inputs that can be given to `determineMediaType` should be tested.